modoptions.lua

```diff
@@ -254,12 +254,18 @@ local options = {
 		column	= 1,
 	},
 	{
-		key		= "disable_unit_sharing",
-		name	= "Disable Unit Sharing",
-		desc	= "Disable sharing units and structures to allies",
-		type	= "bool",
-		section	= "options_main",
-		def		= false,
+		key     = "unit_sharing_mode",
```

Comment:
You expanded unit sharing to be configurable, that's good. But you also removed a previous modoption that players might expect to be available, consider retaining the old modoption and translating it to "Unit Sharing" (mode disabled/enabled) so things keep compatible. Again, depends on whether anyone uses this frequently

Reply:
This preserves the existing option. It just now defaults to "enabled". "disabled" captures the old checkbox binary.

Reply:
If someone types `!bset modoption disable_unit_sharing 1` (or whatever equivalent) is it still supported? Are there other places that might assume the old modoption name exists (e.g. chobby?), and so on. After we've covered our bases, do we care about compatibility or not? (there might be a reasonable argument for not, and I probably even support it, but I want to see this was considered)

I don't mind removing the old modoption, but if it's removed, make sure everything is covered.

Reply:
I think most of the mod_options are reflective inside chobby. But I will verify in in the chobby code it isn't referenced directly.

Yeah, it was considered, one of the reasons I did this PR is because I viewed this as an elegant solution to the problem and thought somebody else might over complicate it. Checkbox -> dropdown seemed clean. For your example it should now be !bset modoption unit_sharing_mode enabled|disabled|t2cons_only (t2cons_only is kind of clunky now that I'm typing it out)

Reply:
Feel free to resolve once you've considered the implications and denoted why this is not an issue according to you (I also agree it's not an issue if the change is just picking a differently named option, and does not break anything)

Reply:
Ok resolving this as I did not find any direct references to `disable_unit_sharing` inside Chobby.